src/mb_enhanced_cover_art_uploads/index.tsx

```diff
@@ -11,6 +13,17 @@ import { ArtworkTypeIDs } from './providers/base';
 
 import type { CoverArt } from './providers/base';
 
+class BadFileTypeError extends CustomError {
+    url: string | URL
+    fileName: string
+
+    constructor(url: string | URL, fileName: string) {
+        super(`${fileName} has an unsupported file type`);
+        this.url = url;
+        this.fileName = fileName;
+    }
+}
+
```

Comment:
I don't think this class should be in the main module of the userscript, a separate module for custom errors might be a cleaner solution. Not sure if the `ResponseError` classes in the XHR module should be moved as well, they rather suit their module compard to the `BadFileTypeError`.

Reply:
I agree, but I'll be doing much more refactoring in #72, so I say we address that there :)

Reply:
Actually, we could just remove this custom error subclass altogether since we're never actually using it specifically, and just throw a normal `Error`. We're only using the `message` field anyway, and I doubt we'll ever need to do an `instanceof` check on these.

The `ResponseError` classes in XHR have more purpose. Even though we're not currently using them in ECAU, I feel like they could still be useful in other scripts.